commit bea79a71f9f9f80a1586374fd716ca8fe0db9e41
Author: Tim Gardner <tim.gardner@canonical.com>
Date:   Thu Jul 26 13:06:26 2012 -0600

    UBUNTU: SAUCE: (no-up) vxge: Declare MODULE_FIRMWARE usage
    
    Obsolete driver, but this helps get rid of extra firmware. Upstream hates it.
    https://lkml.org/lkml/2012/4/12/401
    
    Cc: Jon Mason <jdmason@kudzu.us>
    Cc: "David S. Miller" <davem@davemloft.net>
    Cc: Joe Perches <joe@perches.com>
    Cc: Jiri Pirko <jpirko@redhat.com>
    Cc: Stephen Hemminger <shemminger@vyatta.com>
    Cc: Paul Gortmaker <paul.gortmaker@windriver.com>
    Cc: netdev@vger.kernel.org
    Signed-off-by: Tim Gardner <tim.gardner@canonical.com>

diff --git a/drivers/net/ethernet/neterion/vxge/vxge-main.c b/drivers/net/ethernet/neterion/vxge/vxge-main.c
index e0993eb..67fafe7 100644
--- a/drivers/net/ethernet/neterion/vxge/vxge-main.c
+++ b/drivers/net/ethernet/neterion/vxge/vxge-main.c
@@ -4212,6 +4212,9 @@ out:
 	return ret;
 }
 
+#define VXGE_PXE_FIRMWARE "vxge/X3fw-pxe.ncf"
+#define VXGE_FIRMWARE "vxge/X3fw.ncf"
+
 static int vxge_probe_fw_update(struct vxgedev *vdev)
 {
 	u32 maj, min, bld;
@@ -4254,9 +4257,9 @@ static int vxge_probe_fw_update(struct vxgedev *vdev)
 			}
 	}
 	if (gpxe)
-		fw_name = "vxge/X3fw-pxe.ncf";
+		fw_name = VXGE_PXE_FIRMWARE;
 	else
-		fw_name = "vxge/X3fw.ncf";
+		fw_name = VXGE_FIRMWARE;
 
 	ret = vxge_fw_upgrade(vdev, fw_name, 0);
 	/* -EINVAL and -ENOENT are not fatal errors for flashing firmware on
@@ -4861,3 +4864,5 @@ vxge_closer(void)
 }
 module_init(vxge_starter);
 module_exit(vxge_closer);
+MODULE_FIRMWARE(VXGE_PXE_FIRMWARE);
+MODULE_FIRMWARE(VXGE_FIRMWARE);
